include/avif/internal.h

```diff
@@ -786,6 +786,37 @@ avifResult avifFindMinMaxWithoutOutliers(const float * gainMapF, int numPixels,
 
 #endif // AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP
 
+// ---------------------------------------------------------------------------
+// Internal encode
+//
+// These functions/options give extra flexibility to create non standard images for use in testing.
+
+typedef struct avifEncoderInternalOptions
+{
+#if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
+    // Options related to the 'tmap' (tone mapped image) box.
+    uint8_t tmapVersion;         // Value that should be written for the 'version' field (use default if 0)
+    uint16_t tmapMinimumVersion; // Value that should be written for the 'minimum_version' field (use default if 0)
+    uint16_t tmapWriterVersion;  // Value that should be written for the 'writerVersion' field (use default if 0)
+    avifBool tmapAddExtraBytes;  // Add arbitrary bytes at the end of the box
```

Comment:
Nit: replace "the box" with the name of the class (ToneMapImage or GainMapMetadata?)